inference-engine/src/mkldnn_plugin/nodes/topk.cpp

```diff
@@ -20,51 +23,53 @@ namespace InferenceEngine {
 namespace Extensions {
 namespace Cpu {
 
+using MKLDNNPlugin::TensorDescCreatorTypes;
+
 class TopKImpl: public ExtLayerBase {
 public:
-    explicit TopKImpl(const CNNLayer* layer) {
+    bool isSupportedOperation(const std::shared_ptr<const ngraph::Node>& op, std::string& errorMessage) noexcept {
         try {
-            if (layer->insData.size() != 2)
-                THROW_IE_EXCEPTION << layer->name << " Incorrect number of input edges!";
-
-            if (layer->outData.size() != 1 && layer->outData.size() != 2)
-                THROW_IE_EXCEPTION << layer->name << " Incorrect number of output edges!";
-
-            if (layer->insData[TOPK_K].lock()->getTensorDesc().getDims().size() > 1)
-                THROW_IE_EXCEPTION << layer->name << " TopKImpl - Index vector should be 1 dimension";
-
-            SizeVector dst_dims = layer->outData[0]->getTensorDesc().getDims();
-            SizeVector src_data_dims = layer->insData[TOPK_DATA].lock()->getTensorDesc().getDims();
-            if (src_data_dims.size() != dst_dims.size())
-                THROW_IE_EXCEPTION << layer->name << " TopKImpl - Incorrect input/output tensor dimension sizes";
-
-            if (layer->outData.size() == 2) {
-                SizeVector dst_idx_dims = layer->outData[TOPK_INDEX]->getTensorDesc().getDims();
-                if (dst_dims.size() != dst_idx_dims.size())
-                    THROW_IE_EXCEPTION << layer->name << " Incorrect output tensor dimension sizes";
-
-                for (size_t i = 0; i < dst_dims.size(); i++) {
-                    if (dst_dims[i] != dst_idx_dims[i])
-                        THROW_IE_EXCEPTION << layer->name << " Input/output tensor dimension mismatch";
-                }
+            auto topKOp = ngraph::as_type_ptr<const ngraph::op::v1::TopK>(op);
+            if (!topKOp) {
+                errorMessage = "Node is not an instance of the TopK operation.";
```

Comment:
Not fully correct wording. We have to specify the opset (e.g. `the v1::TopK operation`)

Reply:
FIxed